### PR TITLE
ref(performance): Use CompactSelect for transaction list filter

### PR DESCRIPTION
**Pull request by vuluongj20**
**Before:**
Screen Shot 2022-07-05 at 1 09 07 PM

**After:**
Screen Shot 2022-07-05 at 1 09 19 PM

### CODE REVIEW

File: static/app/components/discover/transactionsList.tsx [hunk 1]
```diff
@@ -176,7 +175,13 @@ class _TransactionsList extends Component<Props> {
...
       <Fragment>
         <div>
-          <DropdownControl
+          <CompactSelect
+            triggerProps={{prefix: t('Filter'), size: 'xsmall'}}
+            value={selected.value}
+            options={options}
+            onChange={opt => handleDropdownChange(opt.value)}
+          />
+          {/* <DropdownControl
```
Comment (malwilley) [inline static/app/components/discover/transactionsList.tsx:184]: Is there a reason for leaving the old component commented out?

File: tests/js/spec/views/performance/transactionSummary.spec.tsx [hunk 2]
```diff
@@ -606,7 +606,7 @@ describe('Performance > TransactionSummary', function () {
         screen.getByRole('button', {name: 'Filter Slow Transactions (p95)'})
       );
 
-      userEvent.click(screen.getByRole('button', {name: 'Slow Transactions (p95)'}));
+      userEvent.click(screen.getAllByText('Slow Transactions (p95)')[1]);
```
Comment (malwilley) [inline tests/js/spec/views/performance/transactionSummary.spec.tsx:609]: Not going to request a change here (unless you're eager), but I'd like to suggest that we (at some point) add accessible roles/attributes to the `<CompactSelect />` component. Like so: https://www.w3.org/WAI/ARIA/apg/example-index/combobox/combobox-select-only.html

That way we can do this which reads better and will be less prone to breaking in the future:

```js
userEvent.click(screen.getByRole('option', {name: 'Slow Transactions (p95)'}));
```

**Review by malwilley — CHANGES_REQUESTED** [review]

File: static/app/components/discover/transactionsList.tsx (hunk 1 above)
Comment (vuluongj20) [reply static/app/components/discover/transactionsList.tsx]: Oops I forgot to delete this – had not had enough coffee lol 😅

File: tests/js/spec/views/performance/transactionSummary.spec.tsx (hunk 2 above)
Comment (malwilley) [reply tests/js/spec/views/performance/transactionSummary.spec.tsx:609]: Ah, interesting. Thanks for linking those issues!

**Review by malwilley — APPROVED** [review]

**Review by udameli — APPROVED** [review]

**Pull request merged**